langserver/handlers/initialize.go

```diff
@@ -108,6 +108,19 @@ func (lh *logHandler) Initialize(ctx context.Context, params lsp.InitializeParam
 		return serverCaps, nil
 	}
 
+	var excludeModulePaths []string
+	for _, rawPath := range cfgOpts.ExcludeModulePaths {
+		rmPath, err := resolvePath(rootDir, rawPath)
+		if err != nil {
+			jrpc2.ServerPush(ctx, "window/showMessage", &lsp.ShowMessageParams{
+				Type:    lsp.MTWarning,
+				Message: fmt.Sprintf("Ignoring exclude root module path %s: %s", rawPath, err),
+			})
```

Comment:
I'm not sure about the value of this warning being surfaced to the user. I would probably just "downgrade" it to a silent log message if not ignore it entirely.

The worst case scenario is that we load some extra modules (e.g. due to typo) and while this may currently cause some CPU spikes I'd say that's something we have to address either way and warning on this level isn't a solution for that problem.

I assume that some users are likely to put a "preemptive list" of some relative paths which may or may not exist in each project and _not finding_ these paths there would be _expected_, so flooding their IDE with warnings is probably undesired in such case.

Reply:
yes, in fact I agree with you. I just found the same code when handling Static user-provided rootModulepaths, I wonder should we "downgrade" to a log message too ?

Reply:
Good question - I'd still keep it for _absolute_ paths there, as that's much more explicit and the user probably doesn't intend for it to be ignored, but we can avoid the warning for relative paths.

You might ask why not apply the same principle here - and I wouldn't object strongly, but I still think that _not_ excluding something you might have wanted to exclude has far less impact than _not including_ - so that's how I'd draw the line.